sys/sys/malloc.h

```diff
@@ -61,6 +61,7 @@
 #define	M_BESTFIT	0x2000		/* only for vmem, low fragmentation */
 #define	M_EXEC		0x4000		/* allocate executable space */
 #define	M_NEXTFIT	0x8000		/* only for vmem, follow cursor */
+#define	M_MEMORY_ARENA	0x10000		/* only for vmem, the arena allocates virtual memory */
```

Comment:
This is an arena flag, not an allocation flag, so should be in a separate block, can take the value 1, and ideally would have a different namespace. It also belongs in a header under sys/vm, not here, given it's passed to vm_init not malloc.

Reply:
I have been conflicted about the flag. I think it is sensible to have it in vmem.h and to allocate a block in the top bits of the `vmem_init` flags and make sure they do not conflict with the `M_*` flags. I initially split them into two blocks with a separate argument to `vmem_create` but opted to do it this way around because it seemed to be more invasive in terms of diff, and there are a bunch of `M_` flags marked as vmem only.